src/meta/processors/admin/BalancePlan.cpp

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/admin/BalancePlan.h"
+#include <folly/synchronization/Baton.h>
+#include "meta/processors/Common.h"
+
+DEFINE_uint32(task_concurrency, 10, "The tasks number could be invoked simultaneously");
+
+namespace nebula {
+namespace meta {
+
+const std::string kBalancePlanTable = "__b_plan__"; // NOLINT
+
+void BalancePlan::dispatchTasks() {
+    // Key -> spaceID + partID,  Val -> List of task index in tasks_;
+    std::unordered_map<std::pair<GraphSpaceID, PartitionID>, std::vector<int32_t>> partTasks;
+    int32_t index = 0;
+    for (auto& task : tasks_) {
+        partTasks[std::make_pair(task.spaceId_, task.partId_)].emplace_back(index++);
+    }
+    buckets_.resize(std::min(tasks_.size(), (size_t)FLAGS_task_concurrency));
```

Comment:
Maybe use partTasks.size() instead of tasks_.size() will be better?

Reply:
Good point!